astroplan/plots/finder.py

```diff
@@ -0,0 +1,125 @@
+
+    log : bool, optional
+        Take the natural logarithm of the FITS image if `True`.
+        False by default.
+
+    ax : `~matplotlib.axes.Axes` or None, optional.
+        The `~matplotlib.axes.Axes` object to be drawn on.
+        If None, uses the current `~matplotlib.axes.Axes`.
+
+    grid : bool, optional.
+        Grid is drawn if `True`. `False` by default.
+
+    reticle : bool, optional
+        Draw reticle on the center of the FOV if `True`. Default is `False`.
+
+    style_kwargs : dict or `None`, optional.
+        A dictionary of keywords passed into `~matplotlib.pyplot.imshow`
+        to set plotting styles.
+
+    reticle_style_kwargs : dict or `None`, optional
+        A dictionary of keywords passed into `~matplotlib.pyplot.axvline` and
+        `~matplotlib.pyplot.axhline` to set reticle style.
+
+    Returns
+    -------
+    ax : `~matplotlib.axes.Axes`
+        Matplotlib axes with survey image centered on ``target``
+
+    hdu : `~astropy.io.fits.PrimaryHDU`
+        FITS HDU of the retrieved image
+    """
+
+    import matplotlib.pyplot as plt
+    from astroquery.skyview import SkyView
+
+    coord = target if not hasattr(target, 'coord') else target.coord
+    position = coord.icrs
+    coordinates = 'icrs'
+    target_name = None if isinstance(target, SkyCoord) else target.name
+
+    hdu = SkyView.get_images(position=position, coordinates=coordinates,
+                             survey=survey, radius=fov_radius, grid=grid)[0][0]
+    wcs = WCS(hdu.header)
+
+    # Set up axes & plot styles if needed.
+    if ax is None:
+        ax = plt.gca(projection=wcs)
+    if style_kwargs is None:
+        style_kwargs = {}
+    style_kwargs = dict(style_kwargs)
+    style_kwargs.setdefault('cmap', 'Greys')
+    style_kwargs.setdefault('origin', 'lower')
+
+    if log:
+        image_data = np.log(hdu.data)
+    else:
+        image_data = hdu.data
+    ax.imshow(image_data, **style_kwargs)
+
+    # Draw reticle
+    if reticle:
+        pixel_width = image_data.shape[0]
+        inner, outer = 0.03, 0.08
+
+        if reticle_style_kwargs is None:
+            reticle_style_kwargs = {}
+        reticle_style_kwargs.setdefault('linewidth', 2)
+        reticle_style_kwargs.setdefault('color', 'm')
+
+        ax.axvline(x=0.5*pixel_width, ymin=0.5+inner, ymax=0.5+outer,
+                   **reticle_style_kwargs)
+        ax.axvline(x=0.5*pixel_width, ymin=0.5-inner, ymax=0.5-outer,
+                   **reticle_style_kwargs)
+        ax.axhline(y=0.5*pixel_width, xmin=0.5+inner, xmax=0.5+outer,
+                   **reticle_style_kwargs)
+        ax.axhline(y=0.5*pixel_width, xmin=0.5-inner, xmax=0.5-outer,
+                   **reticle_style_kwargs)
+
+    # Labels, title, grid
+    ax.set(xlabel='RA', ylabel='DEC')
+    if target_name is not None:
+        ax.set_title(target_name)
+    ax.grid(grid)
+
+    # Redraw the figure for interactive sessions.
```

Comment:
Is it a good idea to call `ax.figure.canvas.draw()` in plotting functions to trigger a redraw for Figures in interactive sessions?
Does this imply inefficient plotting (i.e. double-draw) somehow?

Is this relevant?
http://matplotlib.org/devdocs/users/whats_new.html#interactive-oo-usage

@bmorris3 – If no-one watching this issue comes up with a good answer within ~ a day you could ping e.g. Michael Droettboom with this question.